services/sqs/src/main/java/software/amazon/awssdk/services/sqs/batchmanager/SqsAsyncBatchManager.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.sqs.batchmanager;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledExecutorService;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.services.sqs.SqsAsyncClient;
+import software.amazon.awssdk.services.sqs.internal.batchmanager.BatchOverrideConfiguration;
+import software.amazon.awssdk.services.sqs.internal.batchmanager.DefaultSqsAsyncBatchManager;
+import software.amazon.awssdk.services.sqs.model.ChangeMessageVisibilityRequest;
+import software.amazon.awssdk.services.sqs.model.ChangeMessageVisibilityResponse;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageRequest;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageResponse;
+import software.amazon.awssdk.services.sqs.model.ReceiveMessageRequest;
+import software.amazon.awssdk.services.sqs.model.ReceiveMessageResponse;
+import software.amazon.awssdk.services.sqs.model.SendMessageRequest;
+import software.amazon.awssdk.services.sqs.model.SendMessageResponse;
+import software.amazon.awssdk.utils.SdkAutoCloseable;
+
+/**
+ * Batch manager for implementing automatic batching with an SQS async client. Create an instance using {@link #builder()}.
+ * <p>
+ * This manager buffers client requests and sends them in batches to the service, enhancing efficiency by reducing the number of
+ * API requests. Requests are buffered until they reach a specified limit or a timeout occurs.
+ */
+@SdkPublicApi
+public interface SqsAsyncBatchManager extends SdkAutoCloseable {
+
+    /**
+     * Creates a builder for configuring and creating an {@link SqsAsyncBatchManager}.
+     *
+     * @return A new builder.
+     */
+    static Builder builder() {
+        return DefaultSqsAsyncBatchManager.builder();
+    }
+
+    /**
+     * Buffers and batches {@link SendMessageRequest}s, sending them as a
+     * {@link software.amazon.awssdk.services.sqs.model.SendMessageBatchRequest}. Requests are grouped by queue URL and override
+     * configuration, and sent when the batch size or timeout is reached.
+     *
+     * @param request The SendMessageRequest to be buffered.
+     * @return CompletableFuture of the corresponding {@link SendMessageResponse}.
+     */
+    default CompletableFuture<SendMessageResponse> sendMessage(SendMessageRequest request) {
```

Comment:
Probably out of scope of this PR, but we should probably add consumer builder overload (can be addressed in separate PR)

Reply:
Good point , will mention it as TODO